ikarus/utils/functionhelper.hh

```diff
@@ -37,5 +39,25 @@ void obtainLagrangeNodePositions(const LV& localView,
   for (auto& nCoord : lagrangeNodeCoords)
     nCoord = localView.element().geometry().global(nCoord);
 }
+/**
+ * \brief A function to obtain the local coordinates of the vertices of an FiniteElement
+ * \ingroup utils
+ * \tparam FiniteElement Type of the finite element
+ * \param fe finite element
+ * \return std::vector of LocalCoordinates of the element
+ */
+template <typename FiniteElement>
+auto getVertexPositions(FiniteElement& fe) {
+  constexpr int dim            = FiniteElement::Traits::mydim;
+  const auto& element          = fe.gridElement();
+  const auto& referenceElement = Dune::referenceElement<double, dim>(element.type());
+  const int numberOfVertices   = referenceElement.size(dim);
+
+  std::vector<typename FiniteElement::GridElement::Geometry::LocalCoordinate> positions;
+  for (auto i : std::views::iota(0, numberOfVertices))
+    positions.push_back(referenceElement.position(i, dim));
+
+  return positions;
```

Comment:
This should return a range view instead of a std::vector. Then it will not allocate an std::vector all the time. You can pass the iota view to an transformed_view and return that. https://en.cppreference.com/w/cpp/ranges/transform_view

Reply:
And the function should be named different. First with the get and the it is still misleading, since I may return the global positions of the vertices. Maybe somethign like `referenceElementVertexPositions`

Reply:
Nice, added